### PR TITLE
Server/Core: drop privileges even if not running as daemon

### PR DESCRIPTION
**Pull request by AlexanderS**
As requested in #253 this let the server to drop privileges even if not running as daemon (but if running as root).

### CODE REVIEW

**Comment by solj** [comment]
@madduck Can you give this a shot when you get a chance? If it works as expected, :+1: from me.

File: src/sbin/bcfg2-info [hunk 1]
```diff
@@ -738,6 +740,11 @@ Bcfg2 client itself.""")
     def _daemonize(self):
         pass
 
+    def _drop_privileges(self):
+        os.setgid(self.setup['daemon_gid'])
+        os.setuid(self.setup['daemon_uid'])
+        print("Dropped privileges to %s:%s" % (os.getuid(), os.getgid()))
```
Comment (stpierre) [inline src/sbin/bcfg2-info]: I don't like this ``print()`` -- it could well interfere with the output of ``bcfg2-info``.

**Comment by stpierre** [comment]
There are three different ways to drop privs in this patch:

1. In ``BuiltinCore``, using ``daemon``;
2. In ``CherryPyCore``, using CherryPy;
3. In ``bcfg2-info``, using ``os.set[ug]id``.

Would it be possible and/or reasonable to make the ``daemon`` implementation the default (i.e., in ``BaseCore`` so that both ``BuiltinCore`` and ``bcfg2-info`` could use the same privilege dropping implementation? I'm not even sure if that'd work, but it'd be nice to limit the differences between ``bcfg2-info`` and the builtin core.

**Comment by AlexanderS** [comment]
The daemon variant is not the default. This also removes the print in bcfg2-info because it now always uses logger.debug in BaseCore.

**Comment by stpierre** [comment]
pylint fixes have been backported to maint, so you should be able to rebase this.

**Comment by AlexanderS** [comment]
Rebased to fix build.

**Pull request merged**